smartcontract/service/native/auth/utils.go

```diff
@@ -19,68 +19,224 @@
 package auth
 
 import (
-	"github.com/ontio/ontology/core/states"
-	"github.com/ontio/ontology/core/store/common"
+	"bytes"
+	"fmt"
+
+	"github.com/ontio/ontology/common/serialization"
+	"github.com/ontio/ontology/smartcontract/event"
 	"github.com/ontio/ontology/smartcontract/service/native"
+	"github.com/ontio/ontology/smartcontract/service/native/utils"
 )
 
 var (
-	RoleF        = []byte{0x01}
-	RoleP        = []byte{0x02}
-	FuncPerson   = []byte{0x03}
-	DelegateList = []byte{0x04}
-	Admin        = []byte{0x05}
+	PreAdmin          = []byte{0x01}
+	PreRoleFunc       = []byte{0x02}
+	PreRoleToken      = []byte{0x03}
+	PreDelegateStatus = []byte{0x04}
+	//PreDelegateList   = []byte{0x04}
 )
 
-//this.contractAddr.Admin
-func GetContractAdminKey(native *native.NativeService, contractAddr []byte) ([]byte, error) {
+//type(this.contractAddr.Admin) = []byte
+func concatContractAdminKey(native *native.NativeService, contractAddr []byte) ([]byte, error) {
 	this := native.ContextRef.CurrentContext().ContractAddress
-	adminKey, err := PackKeys(this[:], [][]byte{contractAddr, Admin})
+	adminKey, err := packKeys(this[:], [][]byte{contractAddr, PreAdmin})
```

Comment:
`packKeys` serialize address in varbytes form, which may cause problem.

Reply:
packKeys won't serialize the field param